palettable/utils.py

```diff
@@ -189,9 +194,14 @@ def get_map(name, reverse=False):
 
         if name_lower not in name_map:
             raise KeyError('Unknown palette name: {0}'.format(name))
+        elif len(names_to_data[name]) < length:
```

Comment:
👍 

Could you also add a test of this exception and the new `is_evenly_spaced` behavior in [test_utils.py](https://github.com/andy-esch/palettable/blob/5a6652d8db8fc3620f0e32ffd4843592b299a60c/palettable/test/test_utils.py)?